artifact/gem5art/artifact/artifact.py

```diff
@@ -168,6 +197,19 @@ def registerArtifact(cls,
 
         data['inputs'] = [i._id for i in inputs]
 
+        data['architecture'] = architecture
+        data['size'] = size
+        data['is_zipped'] = is_zipped
+        data['md5sum'] = md5sum
+        data['url'] = url
+        data['supported_gem5_versions'] = supported_gem5_versions
+        data['version'] = version
+
+        for k, v in kwargs.items():
+            if k in data:
+                raise Exception("Field `{}` is redefined".format(k))
```

Comment:
This can't be, right? Because `**kwargs` is whatever is left over.

I think rather than saying that these extra things have a __str__, you should restrict them to be *only strings*. I.e., the type should be `**kwargs: str`. See https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
Yeah, it would be a TypeError if a keyword arg is redefined.